process/interceptors/singleDataInterceptor.go

```diff
@@ -83,14 +78,17 @@ func (sdi *SingleDataInterceptor) ProcessReceivedMessage(message p2p.MessageP2P,
 		return err
 	}
 
-	if !sdi.dataVerifier.IsForCurrentShard(interceptedData) {
+	shouldProcess := interceptedData.IsForCurrentShard() || sdi.whiteListHandler.IsWhiteListed(interceptedData)
```

Comment:
maybe you can extract interceptedData.IsForCurrentShard() in one variable and add it in a trace below exactly as you have done in multiDataInterceptor

Reply:
done